hack/generator/spec/crossplane.md

```diff
@@ -0,0 +1,71 @@
+# Thoughts on performing provider-azure code generation for Crossplane
+
+The patterns identified below were extrapolated from [the Crossplane VNET spec](https://github.com/crossplane/provider-azure/blob/master/config/crd/network.azure.crossplane.io_virtualnetworks.yaml).
+
+## General notes
+
+### Crossplane has the following static "special" properties
+  1. ClaimReference 
```

Comment:
Yeah that's probably worth noting -- I assume we'd avoid using it in the generator then since for new resources my bet is that those fields shouldn't be there (I think that they had confirmed this in a meeting some time back?)

Reply:
I see where you see `ClaimRef`, `ClassRef`, and `ReclaimPolicy` are all deprecated. Where did you see `ProviderRef` was deprecated... and if it's deprecated is `ProviderConfigRef` as well?

Reply:
Found it